src/Webapi/Dom/Webapi__Dom__Element.res

```diff
@@ -1,29 +1,27 @@
 /* internal, moved out of Impl to reduce unnecessary code duplication */
-let ofNode = (node: Dom.node): option<'a> =>
-  Webapi__Dom__Node.nodeType(node) == Webapi__Dom__Types.Element ? Some(Obj.magic(node)) : None
+%%private(
+  let ofNode = (node: Dom.node): option<'a> =>
+    Webapi__Dom__Node.nodeType(node) == Webapi__Dom__Types.Element ? Some(Obj.magic(node)) : None
+  
+  /*
+    At some point we can drop IE11 support and require constructor names to be available
+   */
+  let asHtmlElement = %raw(`
+    function(element) {
+      if ((window.constructor.name !== undefined && /^HTML\w*Element$/.test(element.constructor.name))
```

Comment:
This method exists for two reasons:
- Calling `HtmlElement` binding methods on an `Element`
- As a basic safety net for the `ofElement` methods used by every module that includes `HtmlElement` bindings

Neither of which, imo, need to care about custom elements. This new code should be just as reliable as the check was before but also work across document/window boundaries. I just wanted to make a positive step forward, not fix every possible edge case.

Reply:
It thought it existed to safely cast Element to HtmlElement. But I guess it would be a weird scenario that some one makes a custom element that inherits from Element that is called HTML..Element. I guess if you want to toggle a class on all your custom elements that inherit from HTMLElement you just need to use unsafe casting since we can never for every scenario make that check safe.
I might be to paranoid about type safety. Just trying to think how this would work in some custom element heavy app like a web components app.

Reply:
Looking into it some more toggling classes should work since both element and html element seems to have that property however things like contentEditable or styles will now no longer work if the foo-item and bar-item are web components. The old check worked fine for all apps that doesn't use iframes I think our use case is the odd ball here so not sure we should potentially break existing apps just to fix issues we have with elements in iframes.

```
document
->Document.querySelectorAll("foo-item,bar-item")
->NodeList.toArray
->Belt.Array.forEach(node => {
  node
  ->Element.ofNode
  ->Belt.Option.flatMap(HtmlElement.ofElement)
  ->Belt.Option.forEach(HtmlElement.setContentEditable(_, True))
})
```

I also found an issue with HtmlElement.ofNode that just doesn't do any checking. https://github.com/tinymce/rescript-webapi/issues/75 it just checks if it's a element then downcasts that to a HtmlElement.